### DISCUSSION TITLE
How do you select what heros are disabled?

### DISCUSSION THREAD

**Discussion by Mortis03** [ochadenas/cpudefense#166] [Q&A]
I try to start a game and it says I must select hero a hero to go on leave.  I try to tap on hero's and it says the same thing.  Am I missing something?

**Comment by Mortis03** [comment]
Turns out I have to either return to home screen on android or press recent apps button.  When I return to this app, I can see the heros I can select from.

**Comment by ochadenas** [comment]
This must be version 1.39, right?

Well, how it is _supposed_ to work is that if you enter the intermezzo screen (with the level description), three heroes are presented to choose from, you must select one of them, and only then the two buttons will allow you to go further. This happens for the first time at endless stage 16, and then every four stages. For higher levels it will occur more often, until there will be a hero asking for leave at _each_ stage. Which means that there will be multiple heroes on leave simultaneously.

If you play the level and then return to this screen, you won't be shown the selection dialog. Instead, the game tells you which hero is on leave. Or which heroes are on leave.

If this doesn't work as intended, please tell me exactly what you were doing, perhaps with a screen shot.

Perhaps this is a kind of migration issue, because the feature is newly introduced. But if you continue to think that it's kind of weird behaviour, please say so.

**Comment by Mortis03** [comment]
Thanks for the explanation, I've gotten it to work but I will try to duplicate it and post some screen shots

**Comment by Mortis03** [comment]
Here is a good example:  I tried to start a stage (one I had already heated but didn't get 3 stars) and saw the first screen shot.  After I took the screen shot I cropped it. When I returned to the game I saw the second screen  shot and I was able to select a hero.  


![Screenshot_20240730-083322](https://github.com/user-attachments/assets/bfde3023-9694-46b8-842b-657cd7457bc3)

![Screenshot_20240730-083345](https://github.com/user-attachments/assets/19459052-f9e4-46f1-b17b-08f2f19b0c5f)

**Comment by ochadenas** [comment]
I was not able to reproduce this. But I've added some code that should avoid a situation like this.

I also noticed that although the text says, "will return after 6 levels", the leave duration was always 3 levels. Changed this, too.

**Comment by Mortis03** [comment]
This never really did fix it for me.  Fix was to just switch apps then switch back.  With 1.41 update, switching apps doesn't work anymore.  I cannot select a hero to go on leave.  Also, it says 5 levels.  Is it 3 or 5?

**Reply by ochadenas** [reply]
>  Also, it says 5 levels. Is it 3 or 5?

The text should display the correct number now. So if it says 5, it's 5. The formula is: 1 + (level / 100).

**Comment by ochadenas** [comment]
I have put a test version 1.41.1 here in the "releases" section. Could you have a look and check whether the bug persists?
